tests/test_operational_options.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2020 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA-Commons utilities testing."""
+
+import mock
+import pytest
+
+from reana_commons.errors import REANAValidationError
+from reana_commons.operational_options import validate_operational_options
+
+
+def _mock_check_call(cmd, shell):
+    from reana_commons.operational_options import available_options
+    assert available_options['TARGET']['cwl']() in cmd
+
+
+def test_wrong_option_format():
+    """Test passing option with wrong type."""
+    with pytest.raises(REANAValidationError) as e:
+        workflow_type = 'serial'
+        option_tuple = ('CACHE=on',)
+        validate_operational_options(workflow_type, option_tuple)
+    assert 'must be a dict' in e.value.message
+
+
+def test_unsupported_option():
+    """Test passing unsupported option."""
+    with pytest.raises(REANAValidationError) as e:
+        workflow_type = 'yadage'
+        options = {'unsupported': 'yes'}
+        validate_operational_options(workflow_type, options)
+    assert 'not supported' in e.value.message
+
+
+def test_unsupported_workflow_engine():
+    """Test passing unsupported workflow engine."""
+    with pytest.raises(REANAValidationError) as e:
+        workflow_type = 'foo'
+        options = {'TARGET': 'gendata'}
+        validate_operational_options(workflow_type, options)
+    assert f'not supported for {workflow_type} workflows' in e.value.message
+
+
+def test_unsupported_option_for_workflow_engine():
+    """Test passing unsupported option for valid workflow engine."""
+    with pytest.raises(REANAValidationError) as e:
+        workflow_type = 'serial'
+        options = {'toplevel': 'github:reanahub/awesome-workflow'}
+        validate_operational_options(workflow_type, options)
+    assert f'not supported for {workflow_type} workflows' in e.value.message
```

Comment:
We can write less code here by parametrising the tests https://docs.pytest.org/en/latest/parametrize.html.